src/libraries/Microsoft.Extensions.Logging.Abstractions/src/IBufferedLogRecordMetadata.cs

```diff
@@ -0,0 +1,39 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+
+namespace Microsoft.Extensions.Logging
+{
+    /// <summary>
+    /// Holds metadata for a buffered log record.
+    /// </summary>
+    /// <remarks>
+    /// If the logging infrastructure decides to buffer log records in memory, it will ensure that the
+    /// <c>TState</c> value delivered to logging providers implements this interface. The logging providers
+    /// can then use the metadata to augment the log records they emit.
+    /// </remarks>    
+    public interface IBufferedLogRecordMetadata
+    {
+        /// <summary>
+        /// Gets the time when the log record was recorded.
+        /// </summary>
+        public DateTimeOffset CreationTime { get; }
+
+        /// <summary>
+        /// Gets the ID of the thread that created the log record.
+        /// </summary>
+        public int? ManagedThreadId { get; }
+
+        /// <summary>
+        /// Gets the ID of the span in effect when the log record was created.
+        /// </summary>
+        public ActivitySpanId? SpanId { get; }
+
+        /// <summary>
+        /// Gets the ID of the trace in effect when the log record was created.
+        /// </summary>
+        public ActivityTraceId? TraceId { get; }
+    }
```

Comment:
In the meeting this morning, Noah mentioned another which i think is the TPL GUID for being able to correlate with EventSource tracing?

Reply:
yeah, I thought about it some more and that one is probably obscure enough that its not worth adding now. Some ETW/EventPipe tracing tools use that identifier to associate stuff with async callstacks, but in this case we're not even handling regular old synchronous callstacks yet let alone their async variants. As long as we have some way to add new stuff in the future if needed (default interface method, add properties to a struct) then I think we should be OK.